idx/memory/memory.go

```diff
@@ -206,18 +206,45 @@ func (m *MemoryIdx) Stop() {
 	return
 }
 
-func (m *MemoryIdx) AddOrUpdate(data *schema.MetricData, partition int32) idx.Archive {
+// UpdateMaybe updates an existing archive, if found.
+// it returns the existing archive (if any), and whether it was found
+func (m *MemoryIdx) UpdateMaybe(point schema.MetricPoint, partition int32) (idx.Archive, int32, bool) {
+	pre := time.Now()
+
+	m.Lock()
+	defer m.Unlock()
+
+	existing, ok := m.defById[point.MKey]
+	if ok {
+		oldPart := existing.Partition
+		log.Debug("metricDef with id %v already in index", point.MKey)
```

Comment:
we should probably get rid of log lines that fire on every datapoint received.

a single log.Debug call needs
```
BenchmarkLog-4   	 1000000	      1152 ns/op	      64 B/op	       2 allocs/op
```
https://gist.github.com/woodsaj/33eafd6976181797dfbb46f193f6bcc1

Reply:
OTOH, that's what the debug log calls are for, so you can follow in high detail what happens to a metric at each step along the way. we've historically had these sort of calls in several places (also the input handlers and input plugins).
I agree - obviously - that we should minimize their overhead though. in a few places we guard expensive debug calls with `if LogLevel < 2`, but that's basically to address a shortcoming in our logging library. when we switch to logrus this should no longer be an issue at all (https://github.com/imkira/go-loggers-bench#textnegative-1)

anyway, logging performance tweaks are out of scope for this PR.

Reply:
after a quick look over the code, this is the only place we are calling log.Debug at an insane frequency.  Until we replace the logging lib, lets comment this log line out.  It will literally save us millions of allocations a second across our clusters.

Reply:
it is not. (see kafkaMdm.consumePartition)
but anyway i addressed this in 45c02a0